src/main/java/nl/tudelft/ewi/devhub/server/database/controllers/Deliveries.java

```diff
@@ -74,15 +78,10 @@ public boolean lastDeliveryIsApprovedOrDisapproved(Assignment assignment, Group
                 .list(QDelivery.delivery);
     }
 
-    /**
-     * Get the most recent delivery for every group in this assignment
-     * @param assignment current assignment
-     * @return a list of deliveries
-     */
     @Transactional
-    public List<Delivery> getLastDeliveries(Assignment assignment) {
+    private List<Delivery> getLastDeliveriesList(Assignment assignment, Predicate expression) {
```

Comment:
It does not break existing APIs, as the original getLastDeliveries function still exists. It is merely using this method as helper method. E.g. the content of the method is changed to a private helper method and only the selection of deliveries is delegated to the original API and a new call.